### DISCUSSION TITLE
creating Item Page

### DISCUSSION THREAD

**Discussion by SabSchek** [neuefische/questions#223] [Web] [answered]
Hi there,

I am about to write the code to create a new content page with a selected item from the index page. 
-> this is what I am aiming for 
<img width="215" alt="Bildschirmfoto 2022-10-06 um 11 23 32" src="https://user-images.githubusercontent.com/109827469/194277759-64e7fd0e-3c77-470c-8bca-68baf58aa6d0.png">

So far I wrote a mapping function (definitely with a mistake in it) to be able to click on a certain Itemframe from the index page to 
See it on a new page ->
<img width="399" alt="Bildschirmfoto 2022-10-06 um 11 23 11" src="https://user-images.githubusercontent.com/109827469/194278178-7b45db19-48e4-4eea-aa1b-42698314a6cc.png">
<img width="195" alt="Bildschirmfoto 2022-10-06 um 11 23 16" src="https://user-images.githubusercontent.com/109827469/194278187-681bbf96-112d-4737-9057-0d235eb2910d.png">

I wrote this code here, but can not find the proper code to get to the solution and hope someone can help. 
<img width="895" alt="Bildschirmfoto 2022-10-06 um 11 23 48" src="https://user-images.githubusercontent.com/109827469/194278348-bd647a92-d8b9-4032-b11d-0d887a9357fd.png">

thank you very much in advance x

**Comment by NikRoe** [comment, marked as answer]
Hey fine,
can you send us a PR?

Thanks in advance :)

**Reply by SabSchek** [reply]
https://github.com/SabSchek/capstone-project/pull/15

**Reply by NikRoe** [reply]
You are currently trying to create a component `ItemPage` that display some of the data. You don't need this component.

What you need instead, is a dynamic page that renders the data of the item that is to be displayed. You can choose a similar way as with the filtered goods. 

If you want your component `ItemFrame` to link to a detailed view, you can write it similar to this:
```js
//...
        <Link href={`/product/${id}`} passHref>
          <Button>get it</Button>
        </Link>
//...
```

Afterward you need to work on your dynamic route. It can be similar to this `/pages/product/[id].js`.

In your dynamic route you need a React component that displays the data. It should contain something like this to get the `id` from the URL. This enables you to then figure out which item is to be displayed from your array.
```js
  const router = useRouter();
  const { id } = router.query;

  const filteredGoods = goods.filter((good) => good.id === id);
```
Instead of working with the array method `.filter()`, you can use the [method](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find) `.find()`
Once this is done, you can decide how to render the data.

Hope this helps :)

**Reply by SabSchek** [reply]
Hi Niklas, 
thank you for your help so far. 
I deleted the component ItemPage again and instead it that, I created a file called [id].js in which I implemented the function ItemPage and did import my array of goods and the router. 
<img width="1220" alt="Bildschirmfoto 2022-10-06 um 13 41 59" src="https://user-images.githubusercontent.com/109827469/194304964-2fb48492-2415-436a-b77c-b8cfd5d7cc23.png">

Also I add the Link for the button to the ItemFrane function in ItemFrame.js. 
Unfortunately the local host is still showing an Error on exact that file with the title: Failed to compile.
For me it seems to be a syntax error, the page is saying the same, but I can't find the exact mistake. 
I hope you could have an other look over my code to help me finding the bug. 

Thank you in advance
<img width="1440" alt="Bildschirmfoto 2022-10-06 um 13 44 51" src="https://user-images.githubusercontent.com/109827469/194305484-1c9be0e9-e9b2-4153-b403-7e9e69e1ae4e.png">

**Reply by NikRoe** [reply]
Hey Fine, 
this is a very good start 👌 

You need to ensure that your route in pages matches the route in your `Link` component in your `ItemFrame` component.
Your pages folder currently has a route `/itempages/[id].js` while your `Link` points to `/product/${id}`.

Moreover, your component `ItemFrame` tries to work with a prop `id`, but the component only destructures `name` and `price`.
You need a semicolon after the default export of your `ItemFrame` component and in line 2 of the same file, there is an incomplete import of an `id`. You don't need that import.

In your `ItemPage` component you are using the `.find()`-method correctly, however, it returns the object itself and not an array.
Mapping over the variable `filteredGoods` will thus lead to an error message.
